### DISCUSSION TITLE
Failed to execute goal -checkstyle mvn

### DISCUSSION THREAD

**Discussion by matiashamide** [dds-jv/foro#257] [Consultas de herramientas] [answered]
Buenas, vengo luchando con que el mvn verify no me anda (para el tpi2 unicamente).
funciona mvn test, no me funciona mvn verify y en la consola me devuelve esto:

[INFO] Scanning for projects...
[INFO]
[INFO] -------------------< ar.edu.utn.frba.dds:FileSystem >-------------------
[INFO] Building FileSystem 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- jacoco-maven-plugin:0.8.2:prepare-agent (default-prepare-agent) @ FileSystem ---
[INFO] argLine set to -javaagent:C:\\Users\\Matias\\.m2\\repository\\org\\jacoco\\org.jacoco.agent\\0.8.2\\org.jacoco.agent-0.8.2-runtime.jar=destfile=C:\\Users\\
Matias\\repositorios\\tpi2-sistema-de-archivos-matiashamide\\target\\jacoco.exec
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ FileSystem ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\Users\Matias\repositorios\tpi2-sistema-de-archivos-matiashamide\src\main\resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ FileSystem ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ FileSystem ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\Users\Matias\repositorios\tpi2-sistema-de-archivos-matiashamide\src\test\resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ FileSystem ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- maven-surefire-plugin:2.22.2:test (default-test) @ FileSystem ---
[INFO]
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running fs.BufferTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.035 s - in fs.BufferTest
[INFO] Running fs.HighLevelFileSystemTest
[WARNING] Tests run: 13, Failures: 0, Errors: 0, Skipped: 3, Time elapsed: 0.506 s - in fs.HighLevelFileSystemTest
[INFO]
[INFO] Results:
[INFO]
[WARNING] Tests run: 17, Failures: 0, Errors: 0, Skipped: 3
[INFO]
[INFO]
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ FileSystem ---
[INFO]
[INFO] --- maven-checkstyle-plugin:3.1.2:check (Validar Formateo) @ FileSystem ---
[WARNING] Old version of checkstyle detected. Consider updating to >= v8.30
[WARNING] For more information see: https://maven.apache.org/plugins/maven-checkstyle-plugin/examples/upgrading-checkstyle.html
[INFO] Ignored 40 errors, 302 violations remaining.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[5] (imports) CustomImportOrder: Extra separation in import group before 'java.util.function.Consumer'
[WARNING] src\main\java\fs\ArchivoAbierto.java:[10] (indentation) Indentation: 'member def modifier' has incorrect indentation level 8, expected level should be 2
.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[10,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[11] (indentation) Indentation: 'member def modifier' has incorrect indentation level 8, expected level should be 2
.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[11,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[14] (indentation) Indentation: 'ctor def modifier' has incorrect indentation level 8, expected level should be 2.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[14,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[14,70] (blocks) LeftCurly: '{' at column 63 should have line break after.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[14,95] (coding) OneStatementPerLine: Only one statement per line allowed.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[16] (indentation) Indentation: 'ctor def rcurly' has incorrect indentation level 8, expected level should be 2.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[16,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[18] (indentation) Indentation: 'method def modifier' has incorrect indentation level 8, expected level should be 2
.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[18,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[19] (indentation) Indentation: 'method def' child has incorrect indentation level 16, expected level should be 4.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[19,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[20] (indentation) Indentation: 'method def rcurly' has incorrect indentation level 8, expected level should be 2.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[20,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[22] (indentation) Indentation: 'method def modifier' has incorrect indentation level 8, expected level should be 2
.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[22,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[23] (indentation) Indentation: 'method def' child has incorrect indentation level 16, expected level should be 4.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[23,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[24] (indentation) Indentation: 'method def rcurly' has incorrect indentation level 8, expected level should be 2.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[24,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[26] (indentation) Indentation: 'method def modifier' has incorrect indentation level 8, expected level should be 2
.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[26,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[27] (indentation) Indentation: 'method def' child has incorrect indentation level 16, expected level should be 4.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[27] (sizes) LineLength: Line is longer than 100 characters (found 106).
[WARNING] src\main\java\fs\ArchivoAbierto.java:[27,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[28] (indentation) Indentation: 'method def rcurly' has incorrect indentation level 8, expected level should be 2.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[28,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[30] (indentation) Indentation: 'method def modifier' has incorrect indentation level 8, expected level should be 2
.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[30,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[31] (indentation) Indentation: 'method def' child has incorrect indentation level 16, expected level should be 4.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[31] (sizes) LineLength: Line is longer than 100 characters (found 135).
[WARNING] src\main\java\fs\ArchivoAbierto.java:[31,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[32] (indentation) Indentation: 'method def' child has incorrect indentation level 16, expected level should be 4.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[32] (sizes) LineLength: Line is longer than 100 characters (found 121).
[WARNING] src\main\java\fs\ArchivoAbierto.java:[32,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[33] (indentation) Indentation: 'method def' child has incorrect indentation level 16, expected level should be 4.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[33,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\ArchivoAbierto.java:[34] (indentation) Indentation: 'method def rcurly' has incorrect indentation level 8, expected level should be 2.
[WARNING] src\main\java\fs\LowLevelFileSystem.java:[121,1] (whitespace) FileTabCharacter: Line contains a tab character.
[WARNING] src\main\java\fs\LowLevelFileSystem.java:[122] (sizes) LineLength: Line is longer than 100 characters (found 107).
[WARNING] src\main\java\fs\LowLevelFileSystem.java:[122,1] (whitespace) FileTabCharacter: Line contains a tab character.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.351 s
[INFO] Finished at: 2021-05-29T22:25:26-03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check (Validar Formateo) on project FileSystem: You have 302 Checkstyle viol
ations. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

Entiendo que es un tema de checkstyles que parece que no los estoy cumpliendo pero no se bien donde falla, porque me tira error en todo el codigo fuente. cabe aclarar que le borre muchos warnings para que no se haga tan largo este issue, pero la mayoria son de whitespaces y idention. ya probe reconfigurar lo de los tabs que esta en el README del tpi y aun asi sigo aca :)

cualquier ayuda o consejo me vendria genial, 
desde ya muchas gracias.

**Comment by nanderson87** [comment, marked as answer]
Hola @matiashamide,

Por lo que se ve en los errores que pasaste efectivamente  lo que esta fallando es checkstyle y nuevamente estas en lo cierto en que la mayoría de los errores son de indentación, sangría, etc.

Mi recomendación es la siguiente:
1. Controla que tengas bien para ese proyecto las configuraciones del [IDE](https://github.com/dds-utn/file-system#configuraci%C3%B3n-del-ide-intellij). Más que nada al apartado de tabulación.
2. Y después podes volver intentar darle formato a los archivos donde te esta tirando error.

Siguiendo el siguiente error:

`[WARNING] src\main\java\fs\ArchivoAbierto.java:[10] (indentation) Indentation: 'member def modifier' has incorrect indentation level 8, expected level should be 2 `

Vas al archivo  `ArchivoAbierto`. Abrís el reformant file. Y le das la opción a todo el archivo.

![image](https://user-images.githubusercontent.com/11875266/120090097-4ea67c00-c0d6-11eb-9ef3-c348a33f754f.png)

shortcut para abrir el menu del reformant file segun SO
![image](https://user-images.githubusercontent.com/11875266/120090018-a8f30d00-c0d5-11eb-9a29-01973a82a2fc.png)

Deberías repetir este mismo ejercicio en todos los archivos donde te esta tirando este tipo de errores.

**Reply by matiashamide** [reply]
Buenisimo nico muchas gracias me re sirvio. fui haciendo eso de Ctrl + Alt + L y me fue corrigiendo algunos tabs.
Pero uno de los checkstyle que no me dejaba pasar era el de tenes un paquete que empieze con mayuscula, yo tenia Exceptions y lo tuve que cambiar a excepciones. si a alguno le sucede, fijese tambien de cambiar eso.

**Reply by nanderson87** [reply]
Genial! Y que bueno que pudiste detectar y corregir el otro.

**Reply by FabianaRamirez** [reply]
Hola @matiashamide, ¿cómo estás?

Justo me está pasando lo mismo, pero hay uno de los errores que sigue persistiendo y no se qué hacer.

[INFO] --- maven-checkstyle-plugin:3.1.2:check (Validar Formateo) @ FileSystem ---
[WARNING] Old version of checkstyle detected. Consider updating to >= v8.30
[WARNING] For more information see: https://maven.apache.org/plugins/maven-checkstyle-plugin/examples/upgrading-checkstyle.html
[INFO] Ignored 7 errors, 0 violation remaining.
[INFO] You have 0 Checkstyle violations.

Eso, ¿cómo lo arreglaste?

Desde ya muchas gracias.
Saludos.

**Reply by matiashamide** [reply]
Hola @FabianaRamirez, bien por suerte.
Ese es un warning que tira para que uses la ultima version, pero no es malo ni me tiro error por eso. El warning va a seguir persistiendo a lo largo de la materia

**Reply by FabianaRamirez** [reply]
Hola @matiashamide, muchas gracias por tu respuesta. Hace un ratito lo pude subir y no me tiró error.
Gracias nuevamente. Saludos.
